tower-web/src/app/modules/main/component/sidebar/sidebar.component.ts

```diff
@@ -144,13 +150,24 @@ export class SidebarComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   deleteWorkflow(workflow: Workflow): void {
-    const confirm = prompt(`Please confirm the deletion of the workflow '${workflow.data.runName}' typing its name below (operation is not recoverable):`);
-    if (confirm != workflow.data.runName) {
-      return;
-    }
 
-    this.workflowToDelete = workflow;
-    this.onDeleteWorkflow.next(workflow);
+    this.dialog.open(ConfirmDeleteDialogComponent, {
+      data: {
+        runName: workflow.data.runName
+      },
+      maxWidth: '90%',
+      width: '300px',
```

Comment:
These could also be provided as default using the `MAT_DIALOG_DEFAULT_OPTIONS` DI token if needed & specified by the design system, allowing for less boilerplate.